frontend/packages/erd-core/src/components/ERDRenderer/ERDContent/useUpdateNodeCardinalities.ts

```diff
@@ -0,0 +1,32 @@
+import type { Relationships } from '@liam-hq/db-structure'
+import type { Node } from '@xyflow/react'
+import { useEffect } from 'react'
+import type { TableNodeType } from './TableNode'
+
+export const useUpdateNodeCardinalities = (
+  nodes: Node[],
+  relationships: Relationships,
+) => {
+  useEffect(() => {
+    const hiddenNodes = nodes.filter((n) => n.hidden)
+
+    for (const node of nodes) {
+      const nodeData = node as TableNodeType
+      const tableName = nodeData.data.table.name
+      const targetColumnCardinalities = nodeData.data.targetColumnCardinalities
+
+      if (!tableName || !targetColumnCardinalities) continue
+
+      for (const relationship of Object.values(relationships)) {
+        if (relationship.foreignTableName !== tableName) continue
+
+        const isPrimaryTableHidden = hiddenNodes.some(
+          (hiddenNode) => hiddenNode.id === relationship.primaryTableName,
+        )
+
+        targetColumnCardinalities[relationship.foreignColumnName] =
+          isPrimaryTableHidden ? undefined : relationship.cardinality
```

Comment:
It probably should not be good to be working with this, and even if it is working now, it may break in the future.
You should have to setNodes() here.

https://reactflow.dev/learn/troubleshooting/migrate-to-v12#4-updating-nodes-and-edges

Reply:
I see, I understand we have to use `setNodes()` for updating with new object. Thanks 🙏
a19690c15905bc33885e2a676dc48bb8c99bc8e5